src/raster/gcp.rs

```diff
@@ -1,8 +1,111 @@
 //! Raster ground control point support
 
+use std::ffi::{CStr, CString};
+use std::marker::PhantomData;
+
+use gdal_sys::CPLErr;
+
+use crate::errors::Result;
 use crate::spatial_ref::SpatialRef;
+use crate::utils::{_last_cpl_err, _string};
 use crate::Dataset;
 
+/// An owned Ground Control Point.
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+pub struct Gcp {
+    /// Unique identifier, often numeric.
+    pub id: String,
+    /// Informational message or empty string.
+    pub info: String,
+    /// Pixel (x) location of GCP on raster.
+    pub pixel: f64,
+    /// Line (y) location of GCP on raster.
+    pub line: f64,
+    /// X position of GCP in georeferenced space.
+    pub x: f64,
+    /// Y position of GCP in georeferenced space.
+    pub y: f64,
+    /// Elevation of GCP, or zero if not known.
+    pub z: f64,
+}
```

Comment:
Is there not a way to avoid duplicating the structure of`GDAL_GCP`? Or are the potential options just not worth the effort, or the ergonomics are just bad?

Reply:
Also feels like some impedance mismatching between `Gcp::id` as a field, and `GcpRef::id(&self)` as a function. Can those be made consistent?

Reply:
> Is there not a way to avoid duplicating the structure ofGDAL_GCP? Or are the potential options just not worth the effort, or the ergonomics are just bad?

The easier option would be to add the accessors to GDAL_GCP, but I'm not sure I like mixing types from the two crates (exposing gdal-sys from the gdal API).

Reply:
> Also feels like some impedance mismatching between Gcp::id as a field, and GcpRef::id(&self) as a function. Can those be made consistent?

I guess so, but `GcpRef` is what you consume as a user (so it doesn't really make sense to change it), while `Gcp` is what you set on the dataset, so it's important to be able to create one without a lot of effort. Making the fields private would probably mean something like a builder API.

Reply:
>  not sure I like mixing types from the two crates (exposing gdal-sys from the gdal API).

💯 Agree.